### PR TITLE
separate absolute-time from no-spaces-time

### PR DESCRIPTION
**Pull request by noviluni**
Fixes: https://github.com/scrapinghub/dateparser/issues/751, fixes https://github.com/scrapinghub/dateparser/issues/713, https://github.com/scrapinghub/dateparser/issues/769, reduce false positives in: https://github.com/scrapinghub/dateparser/issues/582

I separated the `no-spaces-time` parser from the `absolute-time` parser. I made some changes in the tests as it was expected to return None when finding invalid inputs but it should return an empty `date_data` (`{'date_obj': None, 'period': ...}`) or raising an exception and I preferred the second approach.

From what I did I thought that we should improve the parsers API, so I opened a new issue: https://github.com/scrapinghub/dateparser/issues/785

### CODE REVIEW

**Review by Gallaecio — APPROVED** [review]

File: dateparser/parser.py [hunk 1]
```diff
@@ -161,11 +155,11 @@ def _find_best_matching_date(cls, datestring):
     @classmethod
     def parse(cls, datestring, settings):
         if not no_space_parser_eligibile(datestring):
-            return
+            raise ValueError('Unable to parse date from: %s' % datestring)
 
         datestring = datestring.replace(':', '')
         if not datestring:
-            return
+            raise ValueError("Empty string")
```
Comment (skyuchukov-bp) [inline dateparser/parser.py:162]: Does this change the current behaviour? Is it a breaking change?

File: dateparser_data/settings.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
     'relative-time',
     'custom-formats',
     'absolute-time',
+    'no-spaces-time',
```
Comment (skyuchukov-bp) [inline dateparser_data/settings.py:6]: This should probably be documented in `docs/usage.rst` (the other parsers are listed there).

Comment (noviluni) [reply dateparser_data/settings.py]: You are totally right! I added it!

File: dateparser/parser.py (hunk 1 above)
Comment (noviluni) [reply dateparser/parser.py:162]: As explained, in the PR description: "I made some changes in the tests as it was expected to return None when finding invalid inputs but it should return an empty date_data ({'date_obj': None, 'period': ...}) or raising an exception and I preferred the second approach". This decision was made to make it behave more similar to the `absolute-time` parser.

About the "breaking change", this PR has been tagged as it. The full implementation is a breaking change.

**Comment by noviluni** [comment]
Hey @Gallaecio @skyuchukov-bp , thank you for the feedback!

I added some information in the docs.

On the other hand, as I have been able to play around with only the `no-spaces-time` parser, I've seen that it returns non-expected results in more than 90% of situations, so I decided to exclude the `no-spaces-time` parser from the default parsers. This parser is causing a lot of side-effect issues and excluding it from the default parsers will fix all of them.

Of course, this is a breaking change, but it's ok as we are going to release a new major version. People will be able to use it by doing: 

```python
    >>> from dateparser_data.settings import default_parsers
    >>> dateparser.parse('2016020417:10', settings={'PARSERS': default_parsers + ['no-spaces-time']})
```

Let me know what you think about this point.

**Comment by noviluni** [comment]
I also checked the  #582 issue and it fixes a lot of the cases:

For example, for this string (created from the second comment in #582):
```python
text = '32° 34’S and 123° and 6005 and 000 and of 629 and >21 and 3533 and 538400'
```

Last released version:

```python
>>> search_dates(text)                                                      
[('32° 34’S', datetime.datetime(2020, 9, 17, 10, 57, 48, 787596)),
 ('123°', datetime.datetime(1900, 1, 1, 1, 2, 3)),
 ('6005', datetime.datetime(6005, 1, 1, 0, 0)),
 ('000', datetime.datetime(1900, 1, 1, 0, 0)),
 ('629', datetime.datetime(1900, 1, 1, 6, 2, 9)),
 ('>21', datetime.datetime(1900, 1, 21, 0, 0)),
 ('3533', datetime.datetime(3533, 1, 21, 0, 0)),
 ('538400', datetime.datetime(8400, 5, 3, 0, 0))]

>>> search_dates(text, settings={'STRICT_PARSING': True})
[('32° 34’S', datetime.datetime(2020, 9, 17, 10, 58, 28, 836995)),
 ('123°', datetime.datetime(1900, 1, 1, 1, 2, 3)),
 ('6005', datetime.datetime(2000, 6, 5, 0, 0)),
 ('000', datetime.datetime(1900, 1, 1, 0, 0)),
 ('629', datetime.datetime(1900, 1, 1, 6, 2, 9)),
 ('>21', datetime.datetime(1900, 1, 1, 2, 1)),
 ('3533', datetime.datetime(2033, 3, 5, 0, 0)),
 ('538400', datetime.datetime(8400, 5, 3, 0, 0))]
```

this PR:

```python
>>> search_dates(text)
[('32° 34’S', datetime.datetime(2020, 9, 17, 11, 4, 21, 207745)),
 ('6005', datetime.datetime(6005, 9, 17, 0, 0)),
 ('>21', datetime.datetime(6005, 9, 21, 0, 0)),
 ('3533', datetime.datetime(3533, 9, 21, 0, 0))]

>>> search_dates(text, settings={'STRICT_PARSING': True})
[('32° 34’S', datetime.datetime(2020, 9, 17, 11, 4, 58, 313809))]
```

**Comment by Gallaecio** [comment]
Looks good to me.

**Comment by noviluni** [comment]
thank you guys!

**Pull request merged**